### ISSUE TITLE
White Vim Vixen Console frame always visible at bottom of GitHub pages

### ISSUE THREAD

**Issue by asampal** [ueokande/vim-vixen#1424]
### I'm opening this issue because:

- [x ] I'll report a bug
- [ ] I'll propose a new feature

### Description

The Vim Vixen iframe with id `vimvixen-console-frame` is always showing, even when the extension is not being actively used, on GitHub pages. The content of the frame is empty/white.
 
#### Steps to Reproduce

-use Firefox Nightly
- go to GitHub site
- observe white area at the bottom of page

![image](https://user-images.githubusercontent.com/281639/169710144-53d0e66d-e01a-4520-b416-2a51cbc0a4c3.png)

#### System configuration

- Operating system: Windows 11
- Firefox version: Nightly 102.0a1
- Vim-Vixen version: 1.2.3

**Comment by jackysee** [comment]
Same here.
It happens to the sites that applied the CSS `color-scheme: dark`.

![vixen-ff](https://user-images.githubusercontent.com/30516/169934766-44a64038-469f-48b8-a576-80d65d4bc3ca.gif)

**Comment by diimdeep** [comment]
Yup, this makes plugin unusable.

**Comment by Vis5** [comment]
+1. I'm currently working around this issue by using Custom CSS add-on. Would love to see it fixed.

**Comment by Aiedails** [comment]
Also for Nightly 103.0a1 on Plasma 5.24.5 Archlinux 5.18.2-arch1-1

**Comment by koveseb** [comment]
I also experience this issue when using the plugin Dark Reader together with Vim Vixen

**Comment by SoundZero** [comment]
I also get the fat white bar at the bottom for many sites, but not all.  I'm using FF Developer Edition 102.0b7 (64-bit).  Seems the "vimvixen-console-frame" iframe is to blame...

**Comment by keyb0ardninja** [comment]
> Just FYI, Firefox 102.0 hit Arch today, and this bug is present there. It's going to affect a lot more users now...

Exactly! Came here after the update.

People familiar with web development might open the dev tools in their browser and figure out the white bar being caused by this particular add-on, but others won't even be able to figure out what's causing the issue.

**Comment by jackysee** [comment]
> +1. I'm currently working around this issue by using Custom CSS add-on. Would love to see it fixed.

@Vis5 what's your work around css?

**Comment by majewsky** [comment]
This workaround CSS works for me:

```css
.vimvixen-console-frame {
  color-scheme: light !important;
}
```

**Comment by AsianKoala** [comment]
Yep just got this after Firefox updated to 102.0 on the AUR.

**Comment by keyb0ardninja** [comment]
> This workaround CSS works for me:
> 
> ```css
> .vimvixen-console-frame {
>   color-scheme: light !important;
> }
> ```

@majewsky

Why `color-scheme: light`? Why not `display: none`:

```css
.vimvixen-console-frame {
  display: none !important;
}
```

**Comment by jackysee** [comment]
@keyb0ardninja  because you still need to use the console

**Comment by keyb0ardninja** [comment]
> @keyb0ardninja because you still need to use the console

Just realized after commenting and deleted my comment before seeing yours 😅

**Comment by InvisibleRain** [comment]
Same on https://scoop.sh/

**Comment by wraiford** [comment]
As a temporary workaround until this gets fixed, it's possible to disable Vim Vixen using the [blacklist fuctionality](https://ueokande.github.io/vim-vixen/blacklist.html) in the extension's preferences:

```json
"blacklist": [
    "github.com/*",
    "*.mozilla.org/*"
]
```

**Comment by pfr-dev** [comment]
This is not just an issue for GitHub pages, I get this on most web pages.

**Comment by jessarcher** [comment]
I think the issue has something to do with https://fvsch.com/transparent-iframes

If I add the following to `src/console/index.html` then it seems to make the iframe transparent again, however, it's briefly still white while the page is loading:

```html
<meta name="color-scheme" content="light dark">
```

**Comment by pfr-dev** [comment]
@jessarcher Thanks for sharing that article, it would seem that is precicely the issue. As a Firefox user, and as the article suggests, I may be stuck until Firefox implements the opaque iframe logic.

**Comment by jessarcher** [comment]
> @jessarcher Thanks for sharing that article, it would seem that is precicely the issue. As a Firefox user, and as the article suggests, I may be stuck until Firefox implements the opaque iframe logic.

I think the problem is that Firefox [*has*](https://twitter.com/fvsch/status/1528722278229413888) implemented the opaque iframe logic. So now, when the website has a CSS `color-scheme` that doesn't match the vim-vixen iframe (currently none) it gives it an opaque background.

I think the trick might be getting the iframe to detect the `color-scheme` of the parent page and then setting its own `color-scheme` to the same value so that Firefox will use a transparent background on the iframe.

**Comment by noobaldrin** [comment]
Can anyone tell me how to install from github? This issue is really annoying.

**Comment by jman-schief** [comment]
> Can anyone tell me how to install from github? This issue is really annoying.

@noobaldrin  The instructions to build and install a fork of this WebExtension with this issue fixed [are at this comment](https://github.com/ueokande/vim-vixen/pull/1437#issuecomment-1176124720). There is no direct link for a ready-made package with the fix. Thanks @jessarcher for authoring #1437.

if you're not comfortable with that, as an alternative you can temporarily install another Firefox add-on and apply the workaround [suggested at this comment](https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1169733896).

**Comment by ouuan** [comment]
> if you're not comfortable with that, as an alternative you can temporarily install another Firefox add-on and apply the workaround [suggested at this comment](https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1169733896).

That's what I use as a workaround now. But why don't you point out what the "another Firefox add-on" is? I don't think it is very helpful without the add-on name. I'm using Stylus.

**Comment by jman-schief** [comment]
> That's what I use as a workaround now. But why don't you point out what the "another Firefox add-on" is? 

good point :) Any Firefox add-on able to override websites CSS is fine (such as Stylus). I randomly picked one and installed [Custom CSS injector](https://addons.mozilla.org/de/firefox/addon/customcss-injector/).

**Comment by pataquets** [comment]
Kudos and thanks to those fixing this! :rocket:
Any ETA for a fix release to be published for those now comfortable with installing outside Mozilla Addons Store? looks quite convoluted... :disappointed:
/cc @ueokande

**Comment by xxSkyy** [comment]
Another implementation of https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1169733896 fix.
1. Download [stylish](https://addons.mozilla.org/en-US/firefox/addon/stylish/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)
2. Install [this style](https://userstyles.org/styles/245566/vim-vixen-temporary-fix)
3. Profit

**Comment by hrqmonteiro** [comment]
> Another implementation of [#1424 (comment)](https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1169733896) fix.
> 
>     1. Download [stylish](https://addons.mozilla.org/en-US/firefox/addon/stylish/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)
> 
>     2. Install [this style](https://userstyles.org/styles/245566/vim-vixen-temporary-fix)
> 
>     3. Profit

It literally doesn't fucking work.

Is there any useful solution? This seems to be a really easy thing to fix, wtf.

**Comment by nonetrix** [comment]
> > Another implementation of [#1424 (comment)](https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1169733896) fix.
> > ```
> > 1. Download [stylish](https://addons.mozilla.org/en-US/firefox/addon/stylish/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)
> > 
> > 2. Install [this style](https://userstyles.org/styles/245566/vim-vixen-temporary-fix)
> > 
> > 3. Profit
> > ```
> 
> It literally doesn't fucking work.
> 
> Is there any useful solution? This seems to be a really easy thing to fix, wtf.

There is no reason to be rude about it that won't get you anywhere in life.

**Comment by nonetrix** [comment]
> This workaround CSS works for me:
> 
> ```css
> .vimvixen-console-frame {
>   color-scheme: light !important;
> }
> ```

fixed it for me thanks

**Comment by rtrajano** [comment]
I dumped the fix into a UserScript. I use [Violentmonkey](https://addons.mozilla.org/en-US/firefox/addon/violentmonkey/) so I'm not sure if this UserScript works in Grease Monkey. Works OK for a workaround except that the box is visible for that split second due to rendering.
```javascript
// ==UserScript==
// @name        Vim Vixen Vix
// @namespace   Violentmonkey Scripts
// @match       *://*/*
// @grant       GM_addStyle
// @version     1.0
// @author      -
// @description Fixes white bottom bar on dark themed web pages when using Vim Vixen
// ==/UserScript==
let styleElement = GM_addStyle(".vimvixen-console-frame {color-scheme: light !important;}");
```

**Comment by scfcode** [comment]
The Violentmonkey script above worked for me (at least for GitHub and scoop.sh) - I am on Firefox 102.0.1, macOS

**Comment by aleksandrs-ledovskis** [comment]
Some extra configuration/styles are necessary for *monkey scripts to remove initial Vixen console's white flash upon page load.

1. Add `height: 0px` rule - `GM_addStyle(".vimvixen-console-frame {height: 0px; color-scheme: light !important;}");`;
2. Change script's "Run at" to "document-start".

**Comment by matt-rayner1** [comment]
> > This workaround CSS works for me:
> > ```css
> > .vimvixen-console-frame {
> >   color-scheme: light !important;
> > }
> > ```
> 
> fixed it for me thanks

Sorry noob here where do I apply this fix?

**Comment by nonetrix** [comment]
> > Sorry noob here where do I apply this fix?
> 
> @matt-rayner1 if you scroll back this thread (starting from [this comment](https://github.com/ueokande/vim-vixen/issues/1424#issuecomment-1177196191)), you'll find tips on how to apply the workaround.
> 
> Short version:
> 
>     1. Install a Firefox add-on that allows changing/customizing the CSS of any webpage you load (read the thread for some suggestions on which one)
> 
>     2. Configure that add-on by adding the above mentioned rule

Seems Hrqmonteiro was sorta right although rude. It seems to fix some sites, and not others. Better than nothing for time being however

**Comment by nonetrix** [comment]
Seems Hrqmonteiro was sorta right although rude. It seems to fix some sites, and not others. Better than nothing for time being however

**Comment by aleksandrs-ledovskis** [comment]
@nonetrix Care to provide any examples of sites that don't work with *monkey scripts?

**Comment by aslak01** [comment]
> The Violentmonkey script above worked for me (at least for GitHub and scoop.sh) - I am on Firefox 102.0.1, macOS

Worked for me too in Firefox Developer Edition 104.0b2

**Comment by AlanMejia10** [comment]
While this extension gets fixed I've been using this other extension named [Tridactyl](https://addons.mozilla.org/en-US/firefox/addon/tridactyl-vim/?utm_source=github.com&utm_content=readme.md), it's very similar and it woks pretty well, I'm only sharing it if anyone wants to give it a shot

**Comment by Timu57** [comment]
Hi, first I want to thank every contributor for this awesome project.
The same bug occurs on https://symfony.com/

**Comment by peteromero** [comment]
> While this extension gets fixed I've been using this other extension named [Tridactyl](https://addons.mozilla.org/en-US/firefox/addon/tridactyl-vim/?utm_source=github.com&utm_content=readme.md), it's very similar and it woks pretty well, I'm only sharing it if anyone wants to give it a shot

I'm really sorry to say, but after weeks of no action, this is the best way to go. 

@ueokande 助けてください！

**Comment by liamtimms** [comment]
Yeah, unfortunately, I'm going to also recommend the move to tridactyl after some testing and the pull request to fix this issue not being merged. That extension's settings can be modified to mimic the behavior expected from vim-vixen.

**Comment by noobaldrin** [comment]
> > While this extension gets fixed I've been using this other extension named [Tridactyl](https://addons.mozilla.org/en-US/firefox/addon/tridactyl-vim/?utm_source=github.com&utm_content=readme.md), it's very similar and it woks pretty well, I'm only sharing it if anyone wants to give it a shot
> 
> I'm really sorry to say, but after weeks of no action, this is the best way to go.
> 
> @ueokande 助けてください！

I used to use Tridactyl until firefox moved to version 4 where it started to not work.
Is it any better than vim-vixen now?

**Comment by asampal** [comment]
Is there a way to override some of Tridactyl's shortcuts? YouTube video control shortcuts seem to be taken over by the extension.

**Comment by liamtimms** [comment]
I think it's best to direct more specific questions about tridactyl to https://github.com/tridactyl/tridactyl to avoid cluttering the issue tracker here.

**Comment by peteromero** [comment]
> Is there a way to override some of Tridactyl's shortcuts? YouTube video control shortcuts seem to be taken over by the extension.

just read the manual. nice explanation.
the command you need begins with "b" and ends with "d" ;)

**Comment by InvisibleRain** [comment]
There hasn't been any response for 3 months and still no fork?

**Comment by stekern** [comment]
If you have enabled loading of `userChrome.css` and `userContent.css` in Firefox, you can also fix the bug without installing any additional plugins for modifying CSS by adding the following to your `userContent.css`:

```css
/* Workaround for vim-vixen issue
 * https://github.com/ueokande/vim-vixen/issues/1424
 */
.vimvixen-console-frame {
  height: 0px;
  color-scheme: light !important;
}
```

**Comment by victorz** [comment]
What's taking so long to fix this? How can we help? This is easily the best keyboard shortcut driven navigation add-on for Firefox that I know of, and I've tried quite a few. I would love to help fix this. Point me in the right direction, _please!_

**Comment by hrqmonteiro** [comment]
> What's taking so long to fix this? How can we help? This is easily the best keyboard shortcut driven navigation add-on for Firefox that I know of, and I've tried quite a few. I would love to help fix this. Point me in the right direction, _please!_

Cause the author doesn't give a f*ck about it's users.

**Comment by ajakk** [comment]
> > What's taking so long to fix this? How can we help? This is easily the best keyboard shortcut driven navigation add-on for Firefox that I know of, and I've tried quite a few. I would love to help fix this. Point me in the right direction, _please!_
> 
> Cause the author doesn't give a f*ck about it's users.

This attitude really has no place anywhere in open source. You're welcome to fix it in your own fork.

**Comment by pataquets** [comment]
> > > What's taking so long to fix this? How can we help? This is easily the best keyboard shortcut driven navigation add-on for Firefox that I know of, and I've tried quite a few. I would love to help fix this. Point me in the right direction, _please!_
> > 
> > 
> > Cause the author doesn't give a f*ck about it's users.
> 
> This attitude really has no place anywhere in open source. You're welcome to fix it in your own fork.

Agreed. The author might be busy and has no obligation, and being rude will definitely not help.
Instead, I've switched to Tridactyl two months ago and it works wonderfully (I found it even more customizable). Thanks @AlanMejia10 for the suggestion, BTW!

**Comment by victorz** [comment]
> Agreed. The author might be busy and has no obligation, and being rude will definitely not help.

But is there a PR up? How can we help get the ball rolling on this?


> Instead, I've switched to Tridactyl two months ago and it works wonderfully (I found it even more customizable). Thanks @AlanMejia10 for the suggestion, BTW!

I tried that. Definitely wasn't as smooth as vixen, in my opinion. 😔 I really want to fix this.

**Comment by majewsky** [comment]
> But is there a PR up?

Yes, in the right sidebar under the "Development" section, you can see a link to #1437.

**Comment by zoj613** [comment]
All this time I thought this was caused by Dark Reader since it only ever appeared when activating dark mode. I have been avoiding dark mode for months because of this and I just found this issue now it disappeared after disabling vim-vixen. Are there alternatives since this plugin does not work when on github?

**Comment by noobaldrin** [comment]
> All this time I thought this was caused by Dark Reader since it only ever appeared when activating dark mode. I have been avoiding dark mode for months because of this and I just found this issue now it disappeared after disabling vim-vixen. Are there alternatives since this plugin does not work when on github?

If you're still having this issue then it's unmaintained now and it's time to move on. I'm currently using Tridactyl now. Thanks for this issue, I can now choose Tridactyl again and realize that it's much better.

**Comment by Ultra980** [comment]
> If you have enabled loading of `userChrome.css` and `userContent.css` in Firefox, you can also fix the bug without installing any additional plugins for modifying CSS by adding the following to your `userContent.css`:
> 
> ```css
> /* Workaround for vim-vixen issue
>  * https://github.com/ueokande/vim-vixen/issues/1424
>  */
> .vimvixen-console-frame {
>   height: 0px;
>   color-scheme: light !important;
> }
> ```

THis doesn't work for me...

**Comment by alex27riva** [comment]
> 

Same for me

**Comment by yujikiriki** [comment]
> > If you have enabled loading of `userChrome.css` and `userContent.css` in Firefox, you can also fix the bug without installing any additional plugins for modifying CSS by adding the following to your `userContent.css`:
> > ```css
> > /* Workaround for vim-vixen issue
> >  * https://github.com/ueokande/vim-vixen/issues/1424
> >  */
> > .vimvixen-console-frame {
> >   height: 0px;
> >   color-scheme: light !important;
> > }
> > ```
> 
> This doesn't work for me...

If in Firefox, install this addon: https://addons.mozilla.org/en-US/firefox/addon/css-override/
Then create a new rule in the addon with the css snippet.
The turn on when in github.
Hope it helps.

**Comment by diimdeep** [comment]
@yujikiriki

**Comment by diederikdehaas** [comment]
> > If you have enabled loading of `userChrome.css` and `userContent.css` in Firefox, you can also fix the bug without installing any additional plugins for modifying CSS by adding the following to your `userContent.css`:
> > ```css
> > /* Workaround for vim-vixen issue
> >  * https://github.com/ueokande/vim-vixen/issues/1424
> >  */
> > .vimvixen-console-frame {
> >   height: 0px;
> >   color-scheme: light !important;
> > }
> > ```
> 
> This doesn't work for me...

Make sure you have set ``toolkit.legacyUserProfileCustomizations.stylesheets`` set to ``true`` in ``about:config``

**Comment by Ultra980** [comment]
@yujikiriki thanks, it worked with [Custom CSS Injector](https://addons.mozilla.org/en-US/firefox/addon/customcss-injector/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search), this one has global CSS.

**Comment by HiImTye** [comment]
I made a userstyle to make adding this easy.
https://userstyles.org/styles/258942/vim-vixen-workaround

**Comment by antonioyon** [comment]
I found an interesting workaround that may hold a clue for a potential solution long-term solution. If I use the Find command (`/` forward slash) It shrinks the frame to the correct size. Then cancel (`esc`) gets out of Find mode and keeps the frame at a single line.

**Comment by m20027** [comment]
I also 
![image](https://user-images.githubusercontent.com/85934819/225299712-a58ad6f4-2571-4a4d-a545-2739a1553170.png)
